src/testdir/test_ins_complete.vim

```diff
@@ -3424,4 +3424,71 @@ func Test_complete_append_selected_match_default()
   delfunc PrintMenuWords
 endfunc
 
+function Test_complete_match()
+  set isexpand=.,/,->,abc,/*,_
+  func TestComplete()
+    let [startcol, expandchar] = complete_match()
+    if startcol >= 0
+      let line = getline('.')
+
+      let items = []
+      if expandchar == '/*'
+        let items = ['/** */']
+      elseif expandchar =~ '^/'
+        let items = ['/*! */', '// TODO:', '// fixme:']
+      elseif expandchar =~ '^\.' && startcol < 4
+        let items = ['length()', 'push()', 'pop()', 'slice()']
+      elseif expandchar =~ '^\.' && startcol > 4
+        let items = ['map()', 'filter()', 'reduce()']
+      elseif expandchar =~ '^\abc'
+        let items = ['def', 'ghk']
+      elseif expandchar =~ '^\->'
+        let items = ['free()', 'xfree()']
+      else
+        let items = ['test1', 'test2', 'test3']
+      endif
+
+      call complete(expandchar =~ '^/' ? startcol : startcol + strlen(expandchar), items)
```

Comment:
@przepompownia about the case what you mentioned. complete_match returns client word boundaries. lsp has its own inferred boundaries. Here we need to consider which one to use based on the special considerations of character. A larger col means that triggerchater is not included, otherwise it is included, which means compl_orig_text will be triggercharacter. Which boundary to use as startcol. Not relevant for this PR.